### PR TITLE
CUMULUS 226 Add kinesis -type rule in Cumulus Rules API

### PR DESCRIPTION
**Pull request by jennyhliu**

### CODE REVIEW

File: packages/api/tests/rules_test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const test = require('ava');
+const models = require('../models');
+const { randomString } = require('@cumulus/common/test-utils');
+
+test('create, update and delete a kinesis type rule', async (t) => {
+  process.env.RulesTable = `RulesTable_${randomString()}`;
+  process.env.stackName = 'my-stackName';
+  //process.env.bucket = 'my-bucket';
```
Comment (abarciauskas-bgse) [inline packages/api/tests/rules_test.js]: nit: remove commented code

File: packages/api/models/rules.js [hunk 2]
```diff
@@ -87,29 +120,75 @@ class Rule extends Manager {
       throw err;
     }
 
-    const payload = await Rule.buildPayload(item);
```
Comment (abarciauskas-bgse) [inline packages/api/models/rules.js]: Is there a reason (besides performance) to move this code inside the switch statement?

File: packages/api/models/rules.js [hunk 3]
```diff
@@ -87,29 +120,75 @@ class Rule extends Manager {
...
+      UUID: item.rule.arn,
+      Enabled: item.state === 'ENABLED'
+    };
+    return await aws.lambda().updateEventSourceMapping(params).promise();
+  }
 
-    // save
-    return super.create(item);
+  /**
+   * delete an event source the from kinesis consumer lambda function
```
Comment (abarciauskas-bgse) [inline packages/api/models/rules.js]: Nit: `Deletes an event source from the kinesis consumer lambda function`

File: packages/api/models/rules.js (hunk 2 above)
Comment (jennyhliu) [reply packages/api/models/rules.js]: I think, creating a kinesis rule doesn't trigger the workflow event.

File: packages/api/models/rules.js [hunk 4]
```diff
@@ -26,25 +27,57 @@ class Rule extends Manager {
...
+        await this.addRule(original, payload);
+        break;
+      }
+      case 'kinesis':
+        if (valueUpdated) {
+          await this.deleteKinesisEventSource(original);
+          await this.addKinesisEventSource(original);
+          updated.rule.arn = original.rule.arn;
+        }
+        else this.updateKinesisEventSource(original);
```
Comment (abarciauskas-bgse) [inline packages/api/models/rules.js]: Nit: Formatting, according to [style guide](https://github.com/airbnb/javascript#blocks--cuddled-elses) should be 
```javascript
if (valueUpdated) {
  // code
} else {
  this.updateKinesisEventSource(original);
}
```

Comment (abarciauskas-bgse) [reply packages/api/models/rules.js]: Also, should it have an `await` ?
```javascript
if (valueUpdated) {
  // code
} else {
  await this.updateKinesisEventSource(original);
}
```

Comment (jennyhliu) [reply packages/api/models/rules.js]: yes.  updated

File: packages/api/tests/rules_test.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+'use strict';
```
Comment (abarciauskas-bgse) [inline packages/api/tests/rules_test.js]: Could we rename this file to `test-rules-model.js` ?

File: packages/api/tests/rules_test.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+const test = require('ava');
+const models = require('../models');
+const { randomString } = require('@cumulus/common/test-utils');
+
+test('create, update and delete a kinesis type rule', async (t) => {
```
Comment (abarciauskas-bgse) [inline packages/api/tests/rules_test.js]: Non-blocking: Would prefer if this was broken into (at least) 3 separate tests

File: packages/api/tests/rules_test.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    })
+    .then(async(rule) => {
+      // delete rule
+      await rules.delete(rule);
+    })
+    .finally(() => {
+      // cleanup table
+      models.Manager.deleteTable(process.env.RulesTable);
+    });
+});
```
Comment (abarciauskas-bgse) [inline packages/api/tests/rules_test.js]: We should probably also include a couple tests for the functionality of deleting / creating / updating a kinesis event source - what do you think?

Comment (jennyhliu) [reply packages/api/tests/rules_test.js]: when a kinesis rule is created, updated or deleted, a kinesis event source is created, updated or deleted, so this test includes them.

**Comment by jennyhliu** [comment]
yeah, I think so.  I have the update in my deploy instance, and I included it now.

**Pull request merged**